File: src/main/java/io/junseok/mealmate/domain/board/dto/response/BoardInfo.java
package io.junseok.mealmate.domain.board.dto.response;

import io.junseok.mealmate.domain.board.entity.Board;
import java.time.LocalDateTime;
import lombok.Builder;

@Builder
public record BoardInfo(
    String title,
    String content,
    String email,
    LocalDateTime lastTime,
    LocalDateTime modifyDt,
    long boardId
) {
    public static BoardInfo fromEntity(Board board){
        return BoardInfo.builder()
            .title(board.getTitle())
            .content(board.getContent())
            .email(board.getMember().getEmail())
            .lastTime(board.createDt)
            .modifyDt(board.modifyDt)
            .boardId(board.getBoardId())
            .build();
    }
}

File: src/main/java/io/junseok/mealmate/domain/board/dto/response/BoardInfo.kt
package io.junseok.mealmate.domain.board.dto.response

import io.junseok.mealmate.domain.board.entity.Board
import java.time.LocalDateTime

data class BoardInfo(
    val title: String,
    val content: String,
    val nickname: String,
    val lastTime: LocalDateTime,
    val modifyDt: LocalDateTime,
    val boardId: Long,
    val email: String
)

fun Board.toCreateBoardInfo() = BoardInfo(
    title = this.title,
    content = this.content,
    nickname = this.member.nickname,
    lastTime = this.createDt!!,
    modifyDt = this.modifyDt!!,
    boardId = this.boardId!!,
    email = this.member.email
)
